### ISSUE TITLE
Feature Request: Warn about trailing backslashes in comments (similar to SC2215)

### ISSUE THREAD

**Issue by schaetzc** [koalaman/shellcheck#2132]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

In below scripts a part of a multi-line command was commented out. However, the comment also disabled a backslash at the end of the line and therefore broke the multi-line command into two commands by accident.

```sh
#!/bin/bash
sed -i \
  -e s/a/b/ \
  # -e s/x/y/ \
  /path/to/file
```

or

```sh
#!/bin/bash
sed -i -e s/a/b/ # -e s/x/y/ \
/path/to/file
```

#### Here's what shellcheck currently says:
```
No issues detected!
```
#### Here's what I wanted or expected to see:

It would be very nice if shellcheck gave a warning about trailing backslashes in comments. The warning could be similar to [SC2215](https://github.com/koalaman/shellcheck/wiki/SC2215). Alternatively, SC2215 could be extended by these kind of mistakes.

**Comment by m-ildefons** [comment]
After reading https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html (section 2.2.1) and rule #9 on token recognition further down on that page, I belive this is a bug not in shellcheck, but in the shell (bash in this case, but dash has this behaviour too)

Your first snippet should be interpreted by the shell just the same as:
```
#!/bin/bash
sed -i -e s/a/b/
```

Since the <backslash> right before the <newline> inside the comment should be expanded before tokenizing, joining the lines, the line following the comment should be part of the comment in its entirety. The same applies to the second example.

I've made a PR just for fun, but I'm not really sure, that that behavior is what we actually want.

**Comment by koalaman** [comment]
I merged the PR with some changes! To avoid false positives it only warns about lines where the previous line was continued (see #2146), so it catches the first and most common case, but not the second. Great suggestion, thanks!
